esp/esdllib/esdl_script.cpp

```diff
@@ -2310,6 +2340,327 @@ class CEsdlTransformOperationFunction : public CEsdlTransformOperationWithChildr
+    {
+        unsigned long long ull;
+        if (s_deserializer.deserialize(rawValue, ull) == Deserialization_SUCCESS)
+        {
+            switch (m_mode)
+            {
+            case Append:
+                if (!txSummary->append(m_info.m_name, ull, m_info.m_level, m_info.m_groups))
+                    UWARNLOG("script operation %s did not append '%s=%s'; name is malformed or is in use", m_tagname.str(), m_info.m_name.str(), rawValue);
+                break;
+            case Set:
+                if (!txSummary->set(m_info.m_name, ull, m_info.m_level, m_info.m_groups))
+                    UWARNLOG("script operation %s did not set '%s=%s'; name is malformed or identifies a cumulative timer", m_tagname.str(), m_info.m_name.str(), rawValue);
+                break;
+            default:
+                break;
+            }
+        }
+        else
+            UWARNLOG("script operation %s did not set '%s=%s'; not an unsigned integer", m_tagname.str(), m_info.m_name.str(), rawValue);
+    }
+    void injectDecimal(const char* rawValue, CTxSummary* txSummary) const
+    {
+        double d;
+        if (s_deserializer.deserialize(rawValue, d) == Deserialization_SUCCESS)
+        {
+            switch (m_mode)
+            {
+            case Append:
+                if (!txSummary->append(m_info.m_name, d, m_info.m_level, m_info.m_groups))
+                    UWARNLOG("script operation %s did not append '%s=%s'; name is malformed or is in use", m_tagname.str(), m_info.m_name.str(), rawValue);
+                break;
+            case Set:
+                if (!txSummary->set(m_info.m_name, d, m_info.m_level, m_info.m_groups))
+                    UWARNLOG("script operation %s did not set '%s=%s'; name is malformed or identifies a cumulative timer", m_tagname.str(), m_info.m_name.str(), rawValue);
+                break;
+            default:
+                break;
+            }
+        }
+        else
+            UWARNLOG("script operation %s did not set '%s=%s'; not decimal", m_tagname.str(), m_info.m_name.str(), rawValue);
+    }
+    void injectValue(const char* rawValue, CTxSummary* txSummary)
+    {
+        switch (m_type)
+        {
+        case Text:
+            injectText(rawValue, txSummary);
+            break;
+        case Signed:
+            injectSigned(rawValue, txSummary);
+            break;
+        case Unsigned:
+            injectUnsigned(rawValue, txSummary);
+            break;
+        case Decimal:
+            injectDecimal(rawValue, txSummary);
+            break;
+        default:
+            break;
+        }
+    }
+    virtual bool exec(CriticalSection* crit, IInterface* preparedForAsync, IEsdlScriptContext* scriptContext, IXpathContext* targetContext, IXpathContext* sourceContext) override
+    {
+        IEspContext* espCtx = scriptContext->queryEspContext();
+        CTxSummary* txSummary = (espCtx ? espCtx->queryTxSummary() : nullptr);
+        if (!txSummary)
+            return true;
```

Comment:
At this time, all ESP contexts include a TxSummary. This is just a sanity check in case that should change.

As for recording what would have been recorded if either the context or summary are missing, I think that would be incorrect behavior. I would interpret the absence of a TxSummary as an intentional effort to not have summary output.

In the case of a scripted log agent, which will exist, I would like to be able to run scripts without even an ESP context. An agent will only have a context if it creates one, and it won't be more than a stub if it does. Does summary output make sense in this scenario?